Tests/UI/New UI/BillingForm/View/ExpiryDateViewTests.swift

```diff
@@ -89,4 +89,139 @@ class ExpiryDateViewTests: XCTestCase {
         view.updateExpiryDate(to: "Test")
         XCTAssertFalse(view.style?.error?.isHidden ?? true)
     }
+
+  func testValidFirstDigitInputWith0() {
+    let input = "0"
+    let textField = UITextField()
+    textField.text = ""
+
+    let shouldContinueAdding = view.textField(textField, shouldChangeCharactersIn: NSRange(location: 0, length: 0), replacementString: input)
+    if shouldContinueAdding {
+      textField.text?.append(input)
```

Comment:
can we avoid any logic inside the unit test to only test what is in the code. For example can we just test the function `validateInput` , This and other places below where if appends.

Reply:
`shouldChangeCharactersIn` is returning `Bool` which allow us to add new change to text field. UITextfield should be more tested using UITest